### PR TITLE
Add and configure esLint + prettier

### PR DESCRIPTION
**Pull request by YuliiaOsadchuk**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*.css
```
Comment (b-sht-lionwood) [inline .eslintignore]: remove

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "plugin:react/recommended",
+        "airbnb",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:prettier/recommended"
```
Comment (b-sht-lionwood) [inline .eslintrc.json]: this is deprecated

Comment (b-sht-lionwood) [reply .eslintrc.json]: pls read https://prettier.io/docs/en/integrating-with-linters.html

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        ],
+        "no-shadow": "off",
+        "@typescript-eslint/no-shadow": ["error"],
+        "@typescript-eslint/explicit-function-return-type": [
+            "error",
+            {
+                "allowExpressions": true
+            }
+        ],
+        "max-len": ["warn", { "code": 80 }],
```
Comment (b-sht-lionwood) [inline .eslintrc.json]: use prettier for that

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+node_modules
```
Comment (b-sht-lionwood) [inline .prettierignore]: remove file

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        "import/prefer-default-export": "off",
+        "react/prop-types": "off",
+        "react/function-component-definition": [
+            2,
+            {
+                "namedComponents": "arrow-function",
+                "unnamedComponents": "arrow-function"
+            }
+        ],
+        "no-loss-of-precision": "error",
```
Comment (b-sht-lionwood) [inline .eslintrc.json]: check rules with example, maybe add more:
```
{
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": [
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "plugin:import/recommended",
    "plugin:import/typescript",
    "plugin:react-redux/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2021,
    "sourceType": "module"
  },
  "plugins": ["react", "@typescript-eslint", "filenames"],
  "rules": {
    "import/extensions": [
      "error",
      "ignorePackages",
      {
        "ts": "never",
        "tsx": "never"
      }
    ],
    "eqeqeq": ["warn", "always"],
    "react/prop-types": "off",
    "import/order": [
      "warn",
      {
        "groups": ["builtin", "external", "internal"],
        "pathGroups": [
          {
            "pattern": "react",
            "group": "external",
            "position": "before"
          }
        ],
        "newlines-between": "always",
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": true
        }
      }
    ],
    "react/jsx-curly-brace-presence": [
      "error",
      {
        "props": "never",
        "children": "never"
      }
    ],
    "react/self-closing-comp": [
      "error",
      {
        "component": true,
        "html": true
      }
    ],
    "no-magic-numbers": [
      "error",
      {
        "ignoreArrayIndexes": true,
        "ignoreDefaultValues": true,
        "enforceConst": false,
        "detectObjects": false,
        "ignore": [-1, 0, 1]
      }
    ],
    "@typescript-eslint/no-explicit-any": ["error"],
    "no-console": [
      "error",
      {
        "allow": ["warn", "error"]
      }
    ],
    "no-duplicate-imports": "error",
    "@typescript-eslint/explicit-function-return-type": [
      "error",
      {
        "allowExpressions": true
      }
    ],
    "lines-between-class-members": [
      "error",
      "always",
      {
        "exceptAfterSingleLine": true
      }
    ],
    "no-else-return": "error",
    "no-implicit-coercion": "error",
    "filenames/match-exported": [2, "kebab"],
    "no-param-reassign": [
      "error",
      {
        "props": true,
        "ignorePropertyModificationsFor": ["state"]
      }
    ],
    "import/no-named-as-default": 0
  },
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "overrides": [
    {
      "files": ["src/app/**/*.{ts,tsx}"],
      "excludedFiles": ["src/app/**/*.styled.tsx", "src/app/repositories/__generated__/**"],
      "rules": {
        "filenames/match-regex": [2, "dt-"],
        "@typescript-eslint/naming-convention": [
          "error",
          {
            "selector": ["variable", "function", "interface", "typeAlias"],
            "format": ["camelCase", "UPPER_CASE", "PascalCase"],
            "prefix": ["dt", "Dt", "DT_"],
            "modifiers": ["exported"]
          }
        ]
      }
    }
  ],
  "ignorePatterns": ["**/*.styled.tsx"]
}

```

File: .prettierrc.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (b-sht-lionwood) [inline .prettierrc.json:1]: extend with:

```
{
  "printWidth": 120,
  "singleQuote": true,
  "useTabs": false,
  "tabWidth": 2,
  "semi": true,
  "bracketSpacing": true,
  "trailingComma": "es5",
  "jsxBracketSameLine": true,
  "jsxSingleQuote": true,
  "endOfLine": "auto"
}
```

File: package.json [hunk 7]
```diff
@@ -1,47 +1,57 @@
...
+        "eslint": "7.32.0",
+        "eslint-config-airbnb": "19.0.1",
+        "eslint-config-prettier": "8.3.0",
+        "eslint-import-resolver-typescript": "2.5.0",
+        "eslint-plugin-import": "2.25.3",
+        "eslint-plugin-jsx-a11y": "6.5.1",
+        "eslint-plugin-prettier": "4.0.0",
+        "eslint-plugin-react": "7.27.1",
+        "eslint-plugin-react-hooks": "4.3.0",
+        "husky": "^7.0.4"
```
Comment (b-sht-lionwood) [inline package.json]: fixed version

Comment (b-sht-lionwood) [reply package.json]: check eslint for rule to have fixed versions in packages.json

**Review by b-sht-lionwood — CHANGES_REQUESTED** [review]
remove yarn-lock from projects

File: package.json (hunk 7 above)
Comment (YuliiaOsadchuk) [reply package.json]: I haven't found such rule

**Review by b-sht-lionwood — APPROVED** [review]

**Pull request merged**